element_array_ephys/export/nwb/nwb.py

```diff
@@ -0,0 +1,600 @@
+            Chunks are blocks of disk space where data are stored contiguously and compressed
+        """
+        self.lfp_electrodes_query = lfp_electrodes_query
+        self.electrodes = self.lfp_electrodes_query.fetch("electrode")
+
+        first_record = (
+            self.lfp_electrodes_query & dict(electrode=self.electrodes[0])
+        ).fetch(as_dict=True)[0]
+
+        self.n_channels = len(lfp_electrodes_query)
+        self.n_tt = len(first_record["lfp"])
+        self._dtype = first_record["lfp"].dtype
+
+        super().__init__(buffer_shape=(self.n_tt, 1), chunk_shape=(chunk_length, 1))
+
+    def _get_data(self, selection):
+
+        electrode = self.electrodes[selection[1]][0]
+        return (self.lfp_electrodes_query & dict(electrode=electrode)).fetch1("lfp")[
+            selection[0], np.newaxis
+        ]
+
+    def _get_dtype(self):
+        return self._dtype
+
+    def _get_maxshape(self):
+        return self.n_tt, self.n_channels
+
+
+def add_electrodes_to_nwb(session_key: dict, nwbfile: pynwb.NWBFile):
+    """
+    Add electrodes table to NWBFile. This is needed for any ElectricalSeries, including
+    raw source data and LFP.
+
+    ephys.InsertionLocation -> ElectrodeGroup.location
+
+    probe.Probe::probe -> device.name
+    probe.Probe::probe_comment -> device.description
+    probe.Probe::probe_type -> device.manufacturer
+
+    probe.ProbeType.Electrode::electrode -> electrodes["id_in_probe"]
+    probe.ProbeType.Electrode::y_coord -> electrodes["rel_y"]
+    probe.ProbeType.Electrode::x_coord -> electrodes["rel_x"]
+    probe.ProbeType.Electrode::shank -> electrodes["shank"]
+    probe.ProbeType.Electrode::shank_col -> electrodes["shank_col"]
+    probe.ProbeType.Electrode::shank_row -> electrodes["shank_row"]
+
+    Parameters
+    ----------
+    session_key: dict
+    nwbfile: pynwb.NWBFile
+    """
+    electrodes_query = probe.ProbeType.Electrode * probe.ElectrodeConfig.Electrode
+
+    for additional_attribute in ["shank_col", "shank_row", "shank"]:
+        nwbfile.add_electrode_column(
+            name=electrodes_query.heading.attributes[additional_attribute].name,
+            description=electrodes_query.heading.attributes[
+                additional_attribute
+            ].comment,
+        )
+
+    nwbfile.add_electrode_column(
+        name="id_in_probe", description="electrode id within the probe",
+    )
+
+    for this_probe in (ephys.ProbeInsertion * probe.Probe & session_key).fetch(
+        as_dict=True
+    ):
+        insertion_record = (ephys.InsertionLocation & this_probe).fetch1()
+        if insertion_record:
+            insert_location = json.dumps(
+                {
+                    k: v
+                    for k, v in insertion_record.items()
+                    if k not in ephys.InsertionLocation.primary_key
+                },
+                cls=DecimalEncoder,
+            )
+        else:
+            insert_location = "unknown"
```

Comment:
```suggestion
        try:
          insertion_record = (ephys.InsertionLocation & this_probe).fetch1()
          insert_location = json.dumps({k: v for k, v in insertion_record.items()
                                                             if k not in ephys.InsertionLocation.primary_key},
                                                             cls=DecimalEncoder)
        except:
            insert_location = "unknown"
```

Reply:
Otherwise `insertion_record = (ephys.InsertionLocation & this_probe).fetch1()` will error out if there are zero entries.

Reply:
I prefer `if`/`else` over `try`/`except` whenever possible. In this case you could check for the length of `ephys.InsertionLocation & this_probe`. `try`/`except` can be OK if you catch specific errors, but an unconditional `except` can often become problematic as it will be used even when the error is not the error that was anticipated.